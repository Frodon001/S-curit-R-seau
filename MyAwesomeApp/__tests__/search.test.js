import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../components/View/search/SearchView';

beforeAll(() => {
    jest.mock('@react-native-community/async-storage');
})


it('renders correctly', async () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Function _searchTextInputChanged change text equal', () => {
    let component = renderer.create(<Search />).getInstance();
    let tree = component._searchTextInputChanged('Mésange');
    expect(tree).toEqual('Mésange');
});

it('Function _searchTextInputChanged change text equal', () => {
    let component = renderer.create(<Search />).getInstance();
    let tree = component._searchTextInputChanged('Moineau');
    expect(tree).not.toEqual('Mésange');
});
