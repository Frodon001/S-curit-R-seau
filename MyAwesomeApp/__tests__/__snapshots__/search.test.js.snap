// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "paddingTop": 5,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Nom de l'oiseau"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderRadius": 10,
          "borderWidth": 1.5,
          "height": 50,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingLeft": 10,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Rechercher
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 20,
            "justifyContent": "center",
            "marginTop": "5%",
            "opacity": 0.5,
            "textAlign": "center",
          },
        ]
      }
    >
       Entrez un nom d'oiseau 
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/searchImage.png",
        }
      }
      style={
        Array [
          Object {
            "flex": 8,
            "height": "90%",
            "marginLeft": "5%",
            "opacity": 0.5,
            "resizeMode": "contain",
            "width": "90%",
          },
        ]
      }
    />
  </View>
</View>
`;
